Podcast/ShareEndpointRequests.swift

```diff
@@ -49,35 +50,26 @@ class FetchSharesEndpointRequest: EndpointRequest {
     }
 }
 
-class CreateShareEndpointRequest: EndpointRequest {
+class ModifyShareEndpointRequest: EndpointRequest {
     
-    var userSharedWithIds: [String]
-    var episodeId: String
+    var userSharedWithIds: [String] = []
+    var id: String
     
-    init(episodeId: String, userSharedWithIds: [String]) {
-        self.episodeId = episodeId
-        self.userSharedWithIds = userSharedWithIds
+    init(id: String, userSharedWithIds: [String] = [], method: HTTPMethod) {
+        self.id = id
         super.init()
         
-        httpMethod = .post
-        path = "/shares/\(episodeId)/"
-        var shareString = ""
-        for (i,id) in userSharedWithIds.enumerated() {
-            shareString += i > 0 ? ("," + id) : id
-        }
-        queryParameters = ["sharee_ids": shareString]
-    }
-}
-
-class DeleteShareEndpointRequest: EndpointRequest {
-    
-    var shareId: String
-    
-    init(shareId: String) {
-        self.shareId = shareId
-        super.init()
+        httpMethod = method
+        path = "/shares/\(id)/"
         
-        httpMethod = .delete
-        path = "/shares/\(shareId)/"
+        if method == .post {
+            self.userSharedWithIds = userSharedWithIds
+            
+            var shareString = ""
+            for (i,id) in userSharedWithIds.enumerated() {
```

Comment:
I think it would be cleaner here to do `userSharedWithIds.enumerated().forEach {...` but it is a personal preference